azurelinuxagent/common/cgroupapi.py

```diff
@@ -124,17 +124,24 @@ def _foreach_controller(operation, message):
         """
         Executes the given operation on all controllers that need to be tracked; outputs 'message' if the controller
         is not mounted or if an error occurs in the operation
+        :return: Returns a list of error messages or an empty list if no errors occurred
         """
+        errors = []
         mounted_controllers = os.listdir(CGROUPS_FILE_SYSTEM_ROOT)
 
         for controller in CGROUP_CONTROLLERS:
             try:
                 if controller not in mounted_controllers:
-                    logger.warn('Cgroup controller "{0}" is not mounted. {1}'.format(controller, message))
+                    msg = 'Cgroup controller "{0}" is not mounted. {1}'.format(controller, message)
+                    logger.warn(msg)
+                    errors.append(msg)
```

Comment:
this shouldn't be flagged as an error

Reply:
I had a dilemma whether to include that or not since it does point to a mismatch between the expected and actual cgroup state on disk. However, I agree it's not pertinent to the cgroup cleanup task. Will remove then.

Reply:
It is not a mismatch between the expected and actual states. This should have been once, at initialization time instead of each time we iterate over the controllers.